docs/pages/management/dynamic-resources/terraform-provider.mdx

```diff
@@ -22,151 +22,165 @@ This guide will explain how to:
-</Admonition>
-
-### Create the Terraform user
-
-Create a local Teleport user named `terraform`, and a matching role granting the
-necessary permissions for Terraform to manage resources in your cluster.
-
-Put the following content into `terraform.yaml`:
-
-```yaml
-kind: role
-metadata:
-  name: terraform
-spec:
-  allow:
-    db_labels:
-      '*': '*'
-    app_labels:
-      '*': '*'
-    rules:
-      - resources:
-        - app
-        - cluster_auth_preference
-        - cluster_networking_config
-        - db
-        - device
-        - github
-        - login_rule
-        - oidc
-        - okta_import_rule
-        - role
-        - saml
-        - session_recording_config
-        - token
-        - trusted_cluster
-        - user
-        verbs: ['list','create','read','update','delete']
-version: v6
----
-kind: user
-metadata:
-  name: terraform
-spec:
-  roles: ['terraform']
-version: v2
-```
-
-Create the `terraform` user and role.
-
-```code
-$ tctl create terraform.yaml
-```
-
-### Enable impersonation
-
-The `terraform` user cannot log in to get credentials, another user must
-**impersonate** this account in order to request a certificate.
-
-Create a role that enables your user to impersonate the Terraform user. First, paste
-the following YAML document into a file called `terraform-impersonator.yaml`:
-
-```yaml
-kind: role
-version: v6
-metadata:
-  name: terraform-impersonator
-spec:
-  allow:
-    # This impersonate rule will allow any user with this role to impersonate
-    # and generate certificates for the user named "terraform" with a role also
-    # named "terraform".
-    impersonate:
-      users: ['terraform']
-      roles: ['terraform']
-```
-
-Next, create the role:
-
-```code
-$ tctl create terraform-impersonator.yaml
-```
-
-(!docs/pages/includes/add-role-to-user.mdx role="terraform-impersonator"!)
-
-Next, request a signed identity file for the Terraform user:
-
-```code
-$ tctl auth sign --user=terraform --out=terraform-identity
-```
-
-This command should result in one file: `terraform-identity`.
-
-## Step 2/3. Create a Terraform configuration
-
-Paste the following into a file called `main.tf` to define an example user and
-role using Terraform.
-
-<Tabs>
-<TabItem scope={["cloud","team"]} label="Cloud-Hosted">
-```
-(!examples/resources/terraform/terraform-user-role-cloud.tf!)
-```
-</TabItem>
-<TabItem scope={["oss", "enterprise"]} label="Self-Hosted">
-```
-(!examples/resources/terraform/terraform-user-role-self-hosted.tf!)
-```
-</TabItem>
-</Tabs>
+Terraform needs a signed identity file from the Teleport cluster certificate
+authority to manage resources in the cluster.
+You can create a local Teleport user for this purpose or you can use a 
+[Machine ID](../../machine-id/introduction.mdx) to generate credentials.
+
+If you intend to run Terraform from a CI/CD platform, a Machine ID is often a better 
+option for generating credentials. A Machine ID can provision ephemeral short-lived 
```

Comment:
```suggestion
option for generating credentials.  Machine ID can provision ephemeral short-lived 
```

Machine ID is one of our product lines, so I think this made more sense before. (We would say "Teleport can X", not "A Teleport can X")